### ISSUE TITLE
Problems using rpi as Dante receiver or for RAVENNA (AES67 transmitter works)

### ISSUE THREAD

**Issue by synapse-md** [bondagit/aes67-linux-daemon#54]
Hi,

I've been trying to get the driver and daemon working on a Raspberry Pi 4B/4GB and am having some difficulty.

I'm using the Raspberry Pi OS Lite, to minimize overhead on the device. I've upgraded to Linux kernel 5.15, and your daemon and patched driver seem to build fine.  I did note a few warnings during compilation regarding unused variables and something about a pointer...

The platform test works on the loopback adapter. 

On the eth0 interface, I can access the WebUI. Testing Dante interoperability, I'm able to see the configured ALSA Source and connect it to 2 channels on an AVIO AES3 adapter. The behavior running speaker-test is correct: prohibition icons get replaced with green icons when there's a signal.

However, the Dante device (set for AES67 interop), does not show up in the WebUI browser.

Also, I can't see any input/output channels for the Linux RAVENNA device in the ANEMAN browser from Merging. The device itself does show up.

Any thoughts on what I'm doing wrong, or advice on how to troubleshoot?

Thanks,
Neil.

**Comment by bondagit** [comment]
It works for me and I am using a Dante AVIOUSB.
The issue caanot be with the driver since SAP. handling is implemented in the daemon.
Can you use a network sniffer and see if you see the Dante SAP announcements? 
They should be sent to 239.255.255.255 port 9875.

**Comment by synapse-md** [comment]
Thanks for this advice.

Interestingly, I can see the SAP announcements from the Daemon, the mDNS announcements from the Daemon, and the mDNS announcements from the Dante AVIO. There are no SAP announcements from the AVIO.

To make sure this isn't a problem with the SAP packetes being passed between my switches, I'm sniffing packets on a PC that hangs from a different switch than the AVIO and Pi. The AVIO and Pi are on the PoE switch. Both the PC switch and the PoE switch cascade from a single switch.

The AVIO is definitely in AES67 mode. Any thoughts on why it might not be making SAP announcements?

FWIW, I'm not using ptp4l -- the AVIO is serving as clock master. The Daemon shows that it's locked to the PTP signal.

**Comment by synapse-md** [comment]
Figured it out! Thanks for the idea of using a sniffer.

I realized that the AVIO was not sending SAP announcements. Looking through the Dante Controller options, I came across this:

![2021-12-03 (1)](https://user-images.githubusercontent.com/84821383/144632768-6a0a5445-3e5d-4d85-84ae-a452d3d65d90.png)

This had to be performed manually to create the flows for announcement via SAP. Now, the AVIO channels show up in the Daemon WebUI Browser.

This may be helpful to add to your interop test instructions.
